### PR TITLE
Make sure not to generate the compile_commands.json when not asked

### PR DESCRIPTION
**Pull request by hakro**
Did not create an issue for this, as the fix is pretty simple.

Scons seems to always generate the `compile_commands.json`, regardless of whether the flag `compiledb` is set or not, and even with an explicit `compiledb=no`.

There are a whopping 3-4 seconds out of 6-10 seconds of build time saved on my machine when I don't need to recreate that file, which is pretty nice when iterating fast on small changes.

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -1015,7 +1015,7 @@ if env["compiledb"] and env.scons_version < (4, 0, 0):
     # Generating the compilation DB (`compile_commands.json`) requires SCons 4.0.0 or later.
     print_error("The `compiledb=yes` option requires SCons 4.0 or later, but your version is %s." % scons_raw_version)
     Exit(255)
-if env.scons_version >= (4, 0, 0):
+if env["compiledb"] and env.scons_version >= (4, 0, 0):
```
Comment (akien-mga) [inline SConstruct:1018]: I'd put this and the previous check in the same `if env["compiledb"]`. I'm pretty this was the case originally but ut must have been broken in a recent refactor.

Comment (hakro) [reply SConstruct:1018]: Sounds good. I merged the 2 blocks.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
I confirm the code was similar in 4.2, so this is a regression fix:
https://github.com/godotengine/godot/blob/3504c98c1233bbd2506e89ce46509bc79afaec17/SConstruct#L952L963

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**